cli/azd/pkg/project/service_config.go

```diff
@@ -45,6 +45,9 @@ type ServiceConfig struct {
 	DotNetContainerApp *DotNetContainerAppOptions `yaml:"-,omitempty"`
 	// Custom configuration for the service target
 	Config map[string]any `yaml:"config,omitempty"`
+	// Computed lazily by useDotnetPublishForDockerBuild and cached. This is true when the project
+	// is a dotnet project and there is not an explicit Dockerfile in the project directory.
+	useDotNetPublishForDockerBuild *bool
```

Comment:
The function `useDotnetPublishForDockerBuild` (which is just a package level function in the `project` package) sets it.

Looking at the code and the callers of it - I think I might be able to arrange things such instead of storing the value here we store it in some cache managed by `ContainerHelper` (likely using the `Path` or something as the cache key).  Do you think that's a better strategy, @wbreza?  FWIW, I did try to call out what was going on here in the comment, but perhaps I should have put `useDotNetPublishForDockerBuild` in square brackets like `[useDotNetPublishForDockerBuild]` so it stood out in the comment a bit more.

Reply:
this looks good to me